classes/form/additional_request.php

```diff
@@ -103,116 +103,6 @@ public function definition() {
         $this->add_action_buttons(true, get_string('submit_additional_review', 'local_extension'));
     }
 
-    /**
-     * Validate the parts of the request form for this module
-     *
-     * @param array $data An array of form data
-     * @param array $files An array of form files
-     * @return array of error messages
-     */
-    public function validation($data, $files) {
-        global $USER;
-
-        $errors = parent::validation($data, $files);
-
-        $mform = $this->_form;
-        $request = $this->_customdata['request'];
-        $cmid = $this->_customdata['cmid'];
-        $mod = $request->mods[$cmid];
-
-        $formid = 'due' . $cmid;
-        $due[$formid] = $data[$formid];
-
-        $handler = $mod->handler;
-        $event = $mod->event;
-
-        // Default request_validation checks for dates within one day of the request.
-        $errors += $handler->request_validation($mform, $mod, $data);
-
-        // Validate any duplicate attachments, prevent submission.
-        if (!empty($data['attachments'])) {
-            $itemid = $data['id'];
-            $draftitemid = $data['attachments'];
-
-            $draftcontext = \context_user::instance($USER->id);
-            $usercontext = \context_user::instance($this->_customdata['request']->request->userid);
-
-            $fs = get_file_storage();
-            $draftfiles = $fs->get_area_files($draftcontext->id, 'user', 'draft', $draftitemid, 'id');
-            $oldfiles = $fs->get_area_files($usercontext->id, 'local_extension', 'attachments', $itemid, 'id');
-
-            if (count($draftfiles) > 1) {
-                $oldnames = array();
-
-                foreach ($oldfiles as $file) {
-                    $oldnames[] = $file->get_filename();
-                }
-
-                $duplicates = array();
-
-                foreach ($draftfiles as $file) {
-                    if (in_array($file->get_filename(), $oldnames) && !$file->is_directory()) {
-                        $duplicates[] = $file->get_filename();
-                    }
-                }
-
-                if (!empty($duplicates)) {
-                    $res = null;
-
-                    $wrapped = array_map(function($str) {
-                        return sprintf("\"%s\"", $str);
-                    }, $duplicates);
-
-                    if (count($wrapped) > 1) {
-                        $last = array_pop($wrapped);
-                        $and = get_string('and', 'local_extension');
-                        $res = implode($wrapped, ', ') . ' ' . $and . ' ' . $last;
-                    } else {
-                        $res = $wrapped[0];
-                    }
-
-                    $errors['attachments'] = get_string('form_rule_validate_duplicate_files', 'local_extension' , $res);
-                }
-            }
-        }
-
```

Comment:
I think we can remove most everything up to this, but leave this last bit of validation for max length.